store/index/index.go

```diff
@@ -42,42 +44,42 @@ func StripBucketPrefix(key []byte, bits byte) []byte {
 //
 // The serialized header is:
 // ```text
-//     |         1 byte        |                1 byte               |
-//     | Version of the header | Number of bits used for the buckets |
+//     |         1 byte        |              1 byte             |              4 bytes             |
+//     | Version of the header | Number of bits used for buckets | First index file number          |
 // ```
 type Header struct {
 	// A version number in case we change the header
 	Version byte
 	// The number of bits used to determine the in-memory buckets
 	BucketsBits byte
+	// First index file number
+	FirstFile uint32
```

Comment:
What is `FirstFile` used for? Couldn't you just scan files based on their filename?

Reply:
Oh, is this use to determine, whether you've wrapped around and start using files again?

Reply:
This is used to find the first (oldest) file to begin the scan from.  It is easier to read here instead of sorting a directory listing and applying logic to detect wrap. With that code here, I get a file name and increment the number until I run out of files.

Reply:
That's a good solution for the "wrap" issue. Though you would still need to sort files as the `FirstFile` is only written once per file, so that file might not exist anymore. Am I missing something?

Reply:
... So yes, this is used to determine where to start in the case the file number has wrapped.

Reply:
No need to sort files, because the next name is computed (baseName + fileNum++) and is opened explicitly.  There is no directory walk.  This does mean that there can be no gap in files.  I think that is OK since GC only attacks the oldest items in the index and stops as soon as anything in a file is still in use.  So, only oldest files get GC'd.

From testing, it appeared there was very little improvement in trying to GC beyond where the first entry still in use existed... at least this is true for the indexer where there is good saturation of the bit prefixes.  For other applications that might not be the case.

Reply:
Got it, thanks. Yes, I also wouldn't create gaps.